sdk/go/arvadosclient/arvadosclient.go

```diff
@@ -149,30 +167,36 @@ func (this ArvadosClient) CallRaw(method string, resource string, uuid string, a
 	}
 
 	defer resp.Body.Close()
-	errorText := fmt.Sprintf("API response: %s", resp.Status)
+	return nil, NewRemoteApiServerError(this.ApiServer, resp)
+}
+
+func NewRemoteApiServerError(ServerAddress string, resp *http.Response) RemoteApiServerError {
```

Comment:
Suggest `newRemoteAPIServerError`. (Or is there a reason to export this?)

(Acronyms are conventionally [API, not Api](https://github.com/golang/go/wiki/CodeReviewComments#initialisms))

Seems to me `RemoteAPIServer` is either redundant or possibly incorrect depending on how you define "remote", and naming the type `APIError` would be fine -- this version will look like `arvadosclient.RemoteAPIServerError` to the caller which seems to belabor the point and even incorrectly suggest a 500-like "server error". This type covers errors like "permission denied", which I would describe as an API or transaction error but not a server error.